### ISSUE TITLE
Publish using Javascript PC emulator

### ISSUE THREAD

**Issue by perlun** [chaos4ever/chaos#80]
Fabrice Bellard has done an excellent Javascript-based emulator based on QEMU: http://bellard.org/jslinux/index.html

We should investigate if it's doable to use this with `chaos` also; it would provide an _extremely_ slim and streamlined way to try chaos out straight in the browser. 😄

**Comment by perlun** [comment]
This one is perhaps even more suitable: https://github.com/copy/v86 (Fabrice's isn't FOSS, and he doesn't allow usage elsewhere without his permission).

**Comment by johannesl** [comment]
Fabrice for sure is an amazing and inspiring programmer. Maybe he's willing to release his sources if you ask him nicely? :-)

In any case, +1 on this since it would be nice to have a way to show people who chaos feels without downloading or installing any virtulization software. Might be exactly what people need to find enough motivation for downloading the RPi SD card and running it on real hardware. :-)

**Comment by perlun** [comment]
> Fabrice for sure is an amazing and inspiring programmer. Maybe he's willing to release his sources if you ask him nicely? :-)

Yeah, we could, but it kind of breaks a critical rule in the [DFSG](https://www.debian.org/social_contract), namely point 8:

> License Must Not Be Specific to Debian
>
> The rights attached to the program must not depend on the program's being part of a Debian system. If the program is extracted from Debian and used or distributed without Debian but otherwise within the terms of the program's license, all parties to whom the program is redistributed should have the same rights as those that are granted in conjunction with the Debian system.

Of course, I know very well that we are not Debian 😛 but I still think it's a valid point. It reminds me a  bit about the whole Bitkeeper/etc debacle years ago, where some people in the Linux kernel acquired a special privilege to use Bitkeeper etc.

This is not exactly the same thing, but I still feel that we should work with Fabrice to release his work as FOSS if possible. Or go with v86. I have no idea which one is _better_ from a technical standpoint.

> In any case, +1 on this since it would be nice to have a way to show people who chaos feels without downloading or installing any virtulization software. Might be exactly what people need to find enough motivation for downloading the RPi SD card and running it on real hardware. :-)

Very true. People are quite lazy nowadays 😆 But I mean seriously, people have other expectations than when we started the chaos project back in the 90's.

A perhaps _simpler_ approach than the JS emulator would be to publish a video using https://asciinema.org/ or similar. We could add a movie where we clone the repo, `vagrant up && vagrant ssh` and compile it in a few minutes; that would be pretty nice. We _could_ even run it, but that will quire running it with something like `qemu` in `ncurses`-mode or so, since asciinema only works with terminal programs (not VirtualBox or anything).

**Comment by johannesl** [comment]
I was actually thinking about trying to get him to publish the sources for anyone to use.. :-) In a DFSG-compatible way.

Your second suggestion about ansciinema is actually brilliant. Let's start there! :-)

JR

**Comment by perlun** [comment]
> I was actually thinking about trying to get him to publish the sources for anyone to use.. :-) In a DFSG-compatible way.

Sounds great!

> Your second suggestion about ansciinema is actually brilliant. Let's start there! :-)

Feel free to submit a PR about it, to the web site repo, if you have time. I'll gladly review it.

**Comment by perlun** [comment]
I did a small test with this, pretty cool: it _almost_ worked:

![image](https://cloud.githubusercontent.com/assets/630613/26029597/42989f32-3841-11e7-9515-6e80f10e78c5.png)

It seems like we use CPU features in the x86 architecture that the emulator does not yet support. 😄 If I'm eager, I might submit a GitHub issues about it or so.

**Comment by perlun** [comment]
Blocked because of #91 for the time being.
